File: python/builder_fee.py
"""
Builder Fee Example

Approve and revoke builder fee permissions.

Requires: PRIVATE_KEY environment variable
"""

import os
from hyperliquid_sdk import HyperliquidSDK

private_key = os.environ.get("PRIVATE_KEY")
if not private_key:
    print("Set PRIVATE_KEY environment variable")
    print("Example: export PRIVATE_KEY='0x...'")
    exit(1)

sdk = HyperliquidSDK(private_key=private_key)
print(f"Wallet: {sdk.address}")

# Check approval status (doesn't require deposit)
status = sdk.approval_status()
print(f"Approval status: {status}")

# Approve builder fee (required before trading via QuickNode)
# Note: Requires account to have deposited first
# result = sdk.approve_builder_fee(max_fee="1%")
# print(f"Approve builder fee: {result}")

# Revoke builder fee permission
# result = sdk.revoke_builder_fee()
# print(f"Revoke builder fee: {result}")

print("\nBuilder fee methods available:")
print("  sdk.approve_builder_fee(max_fee='1%', builder=None)")
print("  sdk.revoke_builder_fee(builder=None)")
print("  sdk.approval_status(user=None)")

File: typescript/builder_fee.ts
#!/usr/bin/env npx ts-node
/**
 * Builder Fee Example
 *
 * Approve and revoke builder fee permissions.
 *
 * Requires: PRIVATE_KEY environment variable
 */

import { HyperliquidSDK } from 'hyperliquid-sdk';

async function main() {
  const privateKey = process.env.PRIVATE_KEY;
  if (!privateKey) {
    console.log("Set PRIVATE_KEY environment variable");
    console.log("Example: export PRIVATE_KEY='0x...'");
    process.exit(1);
  }

  const sdk = new HyperliquidSDK(undefined, { privateKey });
  console.log(`Wallet: ${sdk.address}`);

  // Check approval status (doesn't require deposit)
  const status = await sdk.approvalStatus();
  console.log(`Approval status: ${JSON.stringify(status)}`);

  // Approve builder fee (required before trading via QuickNode)
  // Note: Requires account to have deposited first
  // const result = await sdk.approveBuilderFee("1%");
  // console.log(`Approve builder fee: ${JSON.stringify(result)}`);

  // Revoke builder fee permission
  // const result = await sdk.revokeBuilderFee();
  // console.log(`Revoke builder fee: ${JSON.stringify(result)}`);

  console.log("\nBuilder fee methods available:");
  console.log("  sdk.approveBuilderFee(maxFee, builder?)");
  console.log("  sdk.revokeBuilderFee(builder?)");
  console.log("  sdk.approvalStatus(user?)");
}

main().catch(console.error);
